examples/simulated_tank_gazebo/simulated_tank_gazebo.cc

```diff
@@ -53,12 +57,18 @@ bob_main(int, char **)
     // Create a a vector3 message
     gazebo::msgs::Vector3d msg;
     /************************************Gazebo setup end************/
-
+    //Initialize Gazebo camera display if -d arguement supplied
+    GazeboCameraInput cam(node, "/gazebo/default/differential_drive_robot/camera/link/camera/image");
+    Display display(cam);
+    if(argc==2 && std::string("-d").compare(argv[1])==0){
```

Comment:
I'd write this as ``if(argc >= 2 7& strcmp(argv[1], "-d") == 0) {`` and use the (admittedly ugly) C strings API, so you don't have to construct a new ``std::string`` object. You need to ``#include <cstring>`` for this. Note the ``== 0``: ``strcmp`` rather confusingly doesn't return a boolean.